packages/create-cloudflare/src/common.ts

```diff
@@ -34,13 +34,31 @@ import type { C3Args, PagesGeneratorContext } from "types";
 const { name, npm } = detectPackageManager();
 
 export const validateProjectDirectory = (relativePath: string) => {
+	// Validate that the directory is non-existent or empty
 	const path = resolve(relativePath);
 	const existsAlready = existsSync(path);
 	const isEmpty = existsAlready && readdirSync(path).length === 0; // allow existing dirs _if empty_ to ensure c3 is non-destructive
 
 	if (existsAlready && !isEmpty) {
 		return `Directory \`${relativePath}\` already exists and is not empty. Please choose a new name.`;
 	}
+
+	// Ensure the name is valid per the pages schema
+	const projectName = basename(path);
+	const invalidChars = /^- | -$|[^a-z0-9-]/;
```

Comment:
are the `/^- | -$|` here necessary? aren't you checking that in the next regex?
```suggestion
	const invalidChars = /[^a-z0-9-]/;
```

Reply:
Woops, missed this in the refactor. You're correct